### PR TITLE
Fix dashboard package status badges

### LINKED ISSUE #974: Problem: package status is incorrect or missing in Dashboard
**Describe the bug**

The Enduro Dashboard package status field is not showing the correct value on the package list or package detail screen (see screenshots below).  The package list status column is always blank.  The package detail "Workflow Status" and "Started" values are both blank.

**To Reproduce**

1. Process a SIP with Enduro using a3m as the preservation system
2. Log in to the Enduro Dashboard web UI
3. Click on "Packages" in the navigation pane on the left side of the page
4. Click on a package in the list

**Expected behavior**

The package status field should show the correct value.

**Screenshots**

Screenshot from 2024-06-24 11-23-27
Screenshot from 2024-06-24 11-22-49
Screenshot from 2024-06-24 11-23-11

**Additional context**

The package status was working pretty recently (last week?) so this is a recent regression.

### PR DESCRIPTION
**Pull request by djjuhasz**
Fix #974.

### CODE REVIEW

File: internal/api/design/convert.go [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+// sliceToAny copies the elements of a slice with a specific type (e.g.
+// "[]string") to a slice with type []any. See:
+// https://go.dev/doc/faq#convert_slice_of_interface.
+func sliceToAny[T any](s []T) []any {
+	a := make([]any, len(s))
+	for i, v := range s {
+		a[i] = v
+	}
+	return a
+}
```
Comment (jraddaoui) [inline internal/api/design/convert.go]: There is an `EnumNameInterfaces()` method available in the enum template already ([here](https://github.com/artefactual-sdps/enduro/blob/main/hack/make/enums.tmpl#L6-L13)), you should be able to use that function directly.

File: internal/api/design/package_.go [hunk 2]
```diff
@@ -187,7 +188,7 @@ var _ = Service("package", func() {
 })
 
 var EnumPackageStatus = func() {
-	Enum("new", "in progress", "done", "error", "unknown", "queued", "pending", "abandoned")
+	Enum(sliceToAny(enums.PackageStatusNames())...)
```
Comment (jraddaoui) [inline internal/api/design/package_.go]: `Enum(enums.PackageStatusInterfaces()...)`

**Review by jraddaoui — CHANGES_REQUESTED** [review]
Good thing you noticed this @djjuhasz, thanks!

File: internal/api/design/convert.go (hunk 1 above)
Comment (djjuhasz) [reply internal/api/design/convert.go]: Ah, nice! Thanks. :)

File: internal/api/design/package_.go (hunk 2 above)
Comment (djjuhasz) [reply internal/api/design/package_.go:191]: Oops. :blush:

File: internal/enums/pres_action_type.go [hunk 3]
```diff
@@ -2,10 +2,10 @@ package enums
...
 /*
 ENUM(
-Unspecified
-CreateAIP
-CreateAndReviewAIP
-MovePackage
+unspecified
+create aip
+create and review aip
+move package
```
Comment (jraddaoui) [inline internal/enums/pres_action_type.go:8]: Just noticed, these were using hyphens: https://github.com/artefactual-sdps/enduro/pull/975/files#diff-5d31faeec1ed87c9e54c8a9f758ddbd6640bd2e1a116020c6466dfcdeb25ef28L264

Could that be an issue?

Comment (djjuhasz) [reply internal/enums/pres_action_type.go:8]: It's not an issue for the Enduro or the Dashboard. The strings "create-aip", "create-and-review-aip", and "move-package" don't occur anywhere in this branch anymore.  I've tested the Dashboard and the "Create AIP" and "Move package" titles shows up correctly on the package show page.  As far as I know the "Create and Review AIP" workflow is non-functional in Enduro.

image

**Review by jraddaoui — APPROVED** [review]
Thanks!

**Pull request merged**